### ISSUE TITLE
PHP grpc client leaking memory

### ISSUE THREAD

**Issue by dysosmus** [grpc/grpc#9559]
When running the example `route_guide_client.php` a bit of memory are leaking.
It becomes really problematic when a client is running in a worker (or any long running process).
You can see how obvious is the memory leak with that script
```php
<?php
require dirname(__FILE__).'/../vendor/autoload.php';

@include_once dirname(__FILE__).'/route_guide.pb.php';
@include_once dirname(__FILE__).'/route_guide_grpc_pb.php';

$client = new Routeguide\RouteGuideClient('localhost:50051', [
    'credentials' => Grpc\ChannelCredentials::createInsecure(),
]);
$point = new Routeguide\Point();
$point->setLatitude(409146138);
$point->setLongitude(-746188906);
for($i=0;;$i++) {
    $client->GetFeature($point)->wait();
    usleep(30000);
    if($i%1000) {
        echo "~" . (memory_get_usage() * 10e-6). "Mb\n";
    }
}
```

```shell
> grpc/examples/php/route_guide$ php leak.php
~13.90104Mb
~13.90888Mb
~13.9164Mb
~13.92392Mb
...
~20.78888Mb
~20.7964Mb
~20.80392Mb
```

The environment:
```shell
> grpc/examples/php/route_guide$ php -v
PHP 7.0.15-1~dotdeb+8.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15-1~dotdeb+8.1, Copyright (c) 1999-2017, by Zend Technologies
```

```shell
> grpc/examples/php/route_guide$ pecl list
INSTALLED PACKAGES, CHANNEL PECL.PHP.NET:
=========================================
PACKAGE VERSION STATE
grpc    1.1.0   stable
```

It may be related to that issue #8074

**Comment by stanley-cheung** [comment]
Which protobuf implementation / version are you using?

**Comment by marcelcorso** [comment]
I work with @dysosmus and we are using proto3

**Comment by stanley-cheung** [comment]
Which version? Is it 3.1.0? or 3.2.0?

**Comment by marcelcorso** [comment]
Oh oh. I think I only tried with 3.0.2 which is pretty old. Will try with 3.2.0 tomorrow.

**Comment by stanley-cheung** [comment]
So google/protobuf only starts supporting PHP proto3 since 3.1.0. With 3.2.0 being the latest

**Comment by dysosmus** [comment]
I  check and we are using `3.1.0` I'm going to try if `3.2.0` change something.

**Comment by skormel** [comment]
Hi,

We are using grpc 1.1.0 and protoc 3.1.0 under a lumen project and we've serious leaking memory problems (about 8Mb per http request).  This makes impossible to use it under production enviroment.

Any solution to this error?

Best Regards.
Antonio.

**Comment by skormel** [comment]
We've found an intermediate solution using a wrapper written with phpcpp and grpc (c++ version). Now we've zero memory leaks in php.

**Comment by aleks-pulseid** [comment]
This is a real production blocker for us and this issue has been open for a while know, is there a fix coming soon? 

Setup info:
* extension - `grpc 1.2.0 stable`
* package - `"grpc/grpc": "^v1.3.0"`
* protoc - 3.3.0
* PHP - 5.5.38
* Apache - 2.4.25

**Comment by stanley-cheung** [comment]
Sorry for the delay. I am working on PR #11003 which will hopefully address this.

Before the fix, output of the script at the top of this issue:
```
~19.44656Mb
~19.44784Mb
~19.4488Mb
~19.44976Mb
~19.45072Mb
~19.45168Mb
~19.45264Mb
~19.4536Mb
~19.45456Mb
~19.45552Mb
~19.45648Mb
~19.45744Mb
~19.4584Mb
~19.45936Mb
```

After the fix
```
~19.44464Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
~19.44496Mb
```

**Comment by stanley-cheung** [comment]
Closing this. This should be fixed in the `1.3.2` version released today.
